hack/build-test-images.sh

```diff
@@ -27,3 +27,5 @@ buildimage sti_test/sti-fake-user test/integration/images/sti-fake-user
 buildimage sti_test/sti-fake-scripts test/integration/images/sti-fake-scripts
 buildimage sti_test/sti-fake-scripts-no-save-artifacts test/integration/images/sti-fake-scripts-no-save-artifacts
 buildimage sti_test/sti-fake-no-tar test/integration/images/sti-fake-no-tar
+buildimage sti_test/sti-fake-onbuild-source test/integration/images/sti-fake-onbuild-source
+buildimage sti_test/sti-fake-onbuild test/integration/images/sti-fake-onbuild
```

Comment:
You need one image that contains ONBUILD in its dockerfile.

You then need to use that image as the builder image for an STI invocation.  STI automatically determines that the builder image contains ONBUILD and performs a "pre-build" step in which it docker builds an intermediary image to trigger the ONBUILD instructions.

So again, you just need the one builder image, and then test it like any other builder image, verifying that in the output image, it's clear the ONBUILD instruction was run.